common/src/main/java/com/redhat/cloud/notifications/models/event/TestEventHelper.java

```diff
@@ -0,0 +1,123 @@
+    /**
+     * Creates a test action ready to be sent to the engine. It sets the endpoint's UUID and a flag in the context, so
+     * that it can be easily identified.
+     * @param endpointUuid the endpoint UUID that will be set in the context.
+     * @param orgId the org ID for the action.
+     * @return the created action.
+     */
+    public static Action createTestAction(final UUID endpointUuid, final String orgId) {
+        Action testAction = new Action();
+
+        final var context = new com.redhat.cloud.notifications.ingress.Context();
+        context.setAdditionalProperty(TEST_ACTION_CONTEXT_TEST_EVENT, TEST_ACTION_CONTEXT_TEST_EVENT_VALUE);
+        context.setAdditionalProperty(TEST_ACTION_CONTEXT_ENDPOINT_ID, endpointUuid);
+        testAction.setContext(context);
+
+        /*
+         * Create a test event which will be sent along with the action.
+         */
+        Event testEvent = new Event();
+
+        Metadata metadata = new Metadata();
+        metadata.setAdditionalProperty(TEST_ACTION_METADATA_KEY, TEST_ACTION_METADATA_VALUE);
+
+        Payload payload = new Payload();
+        payload.setAdditionalProperty(TEST_ACTION_PAYLOAD_KEY, TEST_ACTION_PAYLOAD_VALUE);
+
+        testEvent.setMetadata(metadata);
+        testEvent.setPayload(payload);
+
+        Recipient recipient = new Recipient();
+        recipient.setUsers(List.of(TEST_ACTION_RECIPIENT));
+
+        /*
+         * Set the rest of the action's members.
+         */
+        testAction.setApplication(TEST_ACTION_APPLICATION);
+        testAction.setBundle(TEST_ACTION_BUNDLE);
+        testAction.setEvents(List.of(testEvent));
+        testAction.setEventType(TEST_ACTION_EVENT_TYPE);
+        testAction.setId(UUID.randomUUID());
+        testAction.setRecipients(List.of(recipient));
+        testAction.setOrgId(orgId);
+        testAction.setTimestamp(LocalDateTime.now(Clock.systemUTC()));
+        testAction.setVersion(TEST_ACTION_VERSION);
+
+        return testAction;
+    }
+
+    /**
+     * Checks if the provided event is part of an integration test requested by
+     * the client to test their endpoint.
+     * @param event the event to check.
+     * @return true if the event is an integration test, false otherwise.
+     */
+    public static boolean isIntegrationTestEvent(final com.redhat.cloud.notifications.models.Event event) {
+        try {
+            final Context context = event.getAction().getContext();
+            final Boolean isTestEventContextFlag = (Boolean) context.getAdditionalProperties().get(TEST_ACTION_CONTEXT_TEST_EVENT);
+
+            return isTestEventContextFlag &&
+                    event.getAction().getBundle().equals(TestEventHelper.TEST_ACTION_BUNDLE) &&
+                    event.getAction().getApplication().equals(TestEventHelper.TEST_ACTION_APPLICATION) &&
+                    event.getAction().getEventType().equals(TestEventHelper.TEST_ACTION_EVENT_TYPE);
+        } catch (final NullPointerException e) {
```

Comment:
I guess an action should always have a bundle, application or event type, and that we simply reject any actions without this information. I wanted to double check: do I remove it then?

Reply:
The bundle, application and event type fields are mandatory in the incoming `Action` indeed. If any of them is missing, we will reject the Kafka message from the `ingress` topic. So there's no need to catch a NPE here.

Reply:
Some tests will cause a NPE, only because they do not test something that can happen in real life. There's an easy way to work around that without catching the NPE:
```java
            return TestEventHelper.TEST_ACTION_BUNDLE.equals(event.getAction().getBundle()) &&
                    TestEventHelper.TEST_ACTION_APPLICATION.equals(event.getAction().getApplication()) &&
                    TestEventHelper.TEST_ACTION_EVENT_TYPE.equals(event.getAction().getEventType());
```

Reply:
Thank you so much! Changed!